### PR TITLE
Add an icon when there are changes to review

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Highlight when there are new changes to review:

Peek 2020-09-15 11-47

### Applicable issues

  - ref #1907

### CODE REVIEW

File: dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(false);
+    act(() => {
+      wrapper.find("div").simulate("click");
+    });
+    wrapper.update();
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
+
+  it("settind default values removes the icon", () => {
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx:49]: Small typo s/settind/setting/

File: dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    const wrapper = mount(
+      <DifferentialTab
+        deploymentEvent="upgrade"
+        deployedValues="foo"
+        defaultValues="foo"
+        appValues="foo"
+      />,
+    );
+    expect(wrapper.find(CdsIcon).prop("hidden")).toBe(true);
+  });
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx:90]: Is there something missing here for hitting the default values (or clearing changes) should have on the icon, when upgrading?

File: dashboard/src/components/DeploymentFormBody/DifferentialTab.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const [oldValues, setOldValues] = useState("");
+  const setNewChangesFalse = () => setNewChanges(false);
+
+  useEffect(() => {
+    if (deploymentEvent === "upgrade") {
+      // If there are already some deployed values (upgrade scenario)
+      // We compare the values from the old release and the new one
+      setOldValues(deployedValues);
+    } else {
+      // If it's a new deployment, we show the different from the default
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/DifferentialTab.tsx:28]: difference

File: dashboard/src/components/DeploymentFormBody/DifferentialTab.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  appValues,
+}: IDifferentialSelectorProps) {
+  const [newChanges, setNewChanges] = useState(false);
+  const [oldValues, setOldValues] = useState("");
+  const setNewChangesFalse = () => setNewChanges(false);
+
+  useEffect(() => {
+    if (deploymentEvent === "upgrade") {
+      // If there are already some deployed values (upgrade scenario)
+      // We compare the values from the old release and the new one
```
Comment (absoludity) [inline dashboard/src/components/DeploymentFormBody/DifferentialTab.tsx:25]: The code looks like it is setting the old values to the previously deployed values... ah, that's what you mean (the use of "old release" got me, thought you meant the previous chart release as opposed to the previously deployed helm release).

**Review by absoludity — APPROVED** [review]

File: dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx (hunk 2 above)
Comment (andresmgot) [reply dashboard/src/components/DeploymentFormBody/DifferentialTab.test.tsx:90]: yes, thats https://github.com/kubeapps/kubeapps/pull/2032/commits/a817abc14f42eb9f92d499ada72abc52a6c48b5b. When clicking the "restore" button, `oldValues` is equal to `appValues` removing the icon.

**Pull request merged**